  cargo:warning= #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
  cargo:warning=  ^~~~~

---

--- a/src/rust/cryptography-cffi/build.rs
+++ b/src/rust/cryptography-cffi/build.rs
@@ -47,16 +47,8 @@ fn main() {
     )
     .unwrap();
     println!("cargo:rustc-cfg=python_implementation=\"{python_impl}\"");
-    let python_includes = run_python_script(
-        &python,
-        "import os; \
-         import setuptools.dist; \
-         import setuptools.command.build_ext; \
-         b = setuptools.command.build_ext.build_ext(setuptools.dist.Distribution()); \
-         b.finalize_options(); \
-         print(os.pathsep.join(b.include_dirs), end='')",
-    )
-    .unwrap();
+    let python_includes = env::var("PYO3_CROSS_INCLUDE_DIR").unwrap();
+    println!("cargo:rerun-if-env-changed=PYO3_CROSS_INCLUDE_DIR");
     let openssl_c = Path::new(&out_dir).join("_openssl.c");
 
     let mut build = cc::Build::new();
